run-tests.php

```diff
@@ -1470,8 +1470,18 @@ function run_all_tests_parallel($test_files, $env, $redir_tested) {
 					kill_children($workerProcs);
 					error("Could not find worker stdout in array of worker stdouts, THIS SHOULD NOT HAPPEN.");
 				}
-				// Some tests have *very* long output and fgets() seems to truncate to 8KiB unless we say we want more
-				while (FALSE !== ($rawMessage = fgets($workerSock, 64 * 1024))) {
+				$rawMessageBuffer = '';
+				while (FALSE !== ($rawMessage = fgets($workerSock))) {
+					// work around fgets truncating things
```

Comment:
This also apparently doesn't work. I bet there's a non-blocking I/O error in here somewhere.